src/lib-zephyr/alloc.c

```diff
@@ -0,0 +1,280 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Not following why we need another Zephyr directory ?  See https://github.com/thesofproject/sof/pull/6619

Reply:
There was an idea - having 2 dirs lib and zephyr-lib/ and providing same services natively / using zephyr. 
#6619  is another idea of same thing

Reply:
moved to zephyr/lib